buildenv/docker/mkdocker.sh

```diff
@@ -0,0 +1,672 @@
+  echo " && tar -xJf python.tar.xz \\"
+  echo " && cd Python-$python_version \\"
+  echo " && ./configure --prefix=/usr/local \\"
+  echo " && make \\"
+  echo " && make install \\"
+  echo " && cd .. \\"
+  echo " && rm -rf python.tar.xz Python-$python_version"
+}
+
+adjust_ldconfig() {
+  echo ""
+  echo "# Run ldconfig to discover newly installed shared libraries."
+  echo "RUN for dir in lib lib64 ; do echo /usr/local/$dir ; done > /etc/ld.so.conf.d/usr-local.conf \\"
+  echo " && ldconfig"
+}
+
+# called as: add_git_remote remote url
+add_git_remote() {
+  echo " && git remote add ${1} ${2} \\"
+  # Like-named (and conflicting) tags exist in multiple repositories: don't fetch any tags.
+  echo " && git config remote.${1}.tagopt --no-tags \\"
+}
+
+create_git_cache() {
+  # The jdk14 remote is fetched first because that repository was subjected to
+  # 'git gc --aggressive --prune=all' before it was first published making it much
+  # smaller than the other jdk repositories. There is a large degree of overlap
+  # among the jdk repositories so a relatively small number of commits must be
+  # fetched from the others.
+  echo ""
+  echo "# Setup a reference repository cache for faster clones in containers."
+  echo "RUN mkdir /home/$user/git_cache \\"
+  echo " && cd /home/$user/git_cache \\"
+  echo " && git init --bare \\"
+  add_git_remote jdk8    https://github.com/ibmruntimes/openj9-openjdk-jdk8.git
+  add_git_remote jdk11   https://github.com/ibmruntimes/openj9-openjdk-jdk11.git
+  add_git_remote jdk14   https://github.com/ibmruntimes/openj9-openjdk-jdk14.git
+  add_git_remote jdknext https://github.com/ibmruntimes/openj9-openjdk-jdk.git
+  add_git_remote omr     https://github.com/eclipse/openj9-omr.git
+  add_git_remote openj9  https://github.com/eclipse/openj9.git
+  echo " && echo Fetching repository cache... \\"
+  echo " && git fetch jdk14 \\"
+  echo " && git fetch --all \\"
+  echo " && echo Shrinking repository cache... \\"
+  echo " && git gc --aggressive --prune=all"
+}
+
+configure_ssh() {
+  echo ""
+  echo "# Configure sshd."
+  echo "RUN mkdir /var/run/sshd \\"
+  echo " && sed -i -e 's|#PermitRootLogin|PermitRootLogin|' \\"
+  echo "           -e 's|#RSAAuthentication.*|RSAAuthentication yes|' \\"
+  echo "           -e 's|#PubkeyAuthentication.*|PubkeyAuthentication yes|' /etc/ssh/sshd_config"
+  echo ""
+  echo "# SSH login fix so user is not kicked off after login."
+  echo "RUN sed -i -e 's|session\s*required\s*pam_loginuid.so|session optional pam_loginuid.so|' /etc/pam.d/sshd"
+  echo ""
+  echo "# Expose SSH port."
+  echo "EXPOSE 22"
+}
+
+print_dockerfile() {
+  print_license
+  preamble
+  install_packages
+  create_user
+if [ $cuda != no ] ; then
+  install_cuda
+fi
+  install_freemarker
+
+  install_compilers
+
+  install_cmake
+  install_python
+
+  adjust_ldconfig
+  configure_ssh
+
+  install_bootjdks
```

Comment:
This seems to download quite a few JDKs. Is the intention that the resulting Dockerfile will be able to build all versions of the JDK? If so does that mean the user is expected to explicitly initialize `JAVA_HOME` to point to the correct JDK before kicking off the build process?

Reply:
Yes, boot JDKs will be available for building any (current) version. Users are expected to use the `--with-boot-jdk=` configure option, e.g.:
```
bash configure --with-boot-jdk=/usr/lib/jvm/adoptojdk-java-80 ...
```
or
```
bash configure --with-boot-jdk=/usr/lib/jvm/adoptojdk-java-11 ...
```
Setting `JAVA_HOME` is not necessary.

Reply:
I use the `--with-boot-jdk=` configure option, but I see we don't consistently suggest that in the build docs, it's only suggested for Windows.

Reply:
Created https://github.com/eclipse/openj9/issues/9375 to deal with this problem.